test/https/https.test.js

```diff
@@ -0,0 +1,53 @@
+'use strict'
+
+const t = require('tap')
+const test = t.test
+const request = require('request')
+const fs = require('fs')
+const path = require('path')
+const Fastify = require('../..')
+
+// Because we are using a self signed certificate for testing
+process.env.NODE_TLS_REJECT_UNAUTHORIZED = '0'
```

Comment:
Good point, but `rejectUnauthorized ` must be `false` because we are using a self-signed certificate, otherwise we'll get a:
```
Error: self signed certificate
at Error (native)
at TLSSocket.<anonymous> (_tls_wrap.js:1079:38)
at emitNone (events.js:86:13)
at TLSSocket.emit (events.js:185:7)
at TLSSocket._finishInit (_tls_wrap.js:603:8)
at TLSWrap.ssl.onhandshakedone (_tls_wrap.js:433:38)
message: self signed certificate
code: DEPTH_ZERO_SELF_SIGNED_CERT
```